analysis_driver/pipelines/demultiplexing.py

```diff
@@ -49,8 +49,13 @@ def demultiplexing_pipeline(dataset):
     crawler.send_data()
     dataset.end_stage('setup')
 
-    # Need to sleep to make sure the LIMS has had enough time to update itself
-    time.sleep(900)
+    validation_log = join(job_dir, 'checked_bcls.csv')
+    b = BCLValidator(job_dir, run_info, validation_log, dataset)
+    b.check_bcls()
+    invalid_bcls = b.read_invalid_files()
+    if invalid_bcls:
+        raise AnalysisDriverError('Some BCL files are corrupted; check %s for details', validation_log)
+
     run_status = clarity.get_run(run_id).udf.get('Run Status')
```

Comment:
we should retrieve the status from the stored Lims object with
```python
if dataset.lims_run.udf.get('Run Status') != 'RunCompleted':
```
also remove the TODO as it is now taken care of